programs/drift/src/math/amm_jit.rs

```diff
@@ -18,9 +19,22 @@ pub fn calculate_jit_base_asset_amount(
     auction_price: u64,
     valid_oracle_price: Option<i64>,
     taker_direction: PositionDirection,
+    split_with_lps: bool,
 ) -> DriftResult<u64> {
-    // only take up to 50% of what the maker is making
-    let mut max_jit_amount = maker_base_asset_amount.safe_div(2)?;
+    let user_lp_ratio = market
+        .amm
+        .user_lp_shares
+        .safe_div(market.amm.sqrt_k.safe_div(100)?)?
+        .cast::<u64>()?;
+
+    // protocol-owend lp only take up to 50% of what the maker is making
```

Comment:
is this still true? if lp ratio is 1, it looks like `maker_base_asset_amount` can be 51%?

Reply:
i think using `amm_lp_allowed_to_jit_make` to control this is better than giving active maker new unexpected results